examples/Basics/user_callable.py

```diff
@@ -0,0 +1,126 @@
+## Example of a user-defined callable
+
+# This example is an extension of the recommender example.
+# It implements an end-to-end use case demonstrating how to utilize the user-defined callable in pure recommenders.
+
+# This examples assumes some basic familiarity with using BayBE.
+# We refer to [`campaign`](./campaign.md) for a more general and basic example.
+
+### Necessary imports for this example
+
+from typing import Optional
+
+from pandas import DataFrame
+
+from baybe import Campaign
+from baybe.objectives import SingleTargetObjective
+from baybe.parameters import NumericalDiscreteParameter, SubstanceParameter
+from baybe.recommenders import (
+    RandomRecommender,
+    SequentialGreedyRecommender,
+    TwoPhaseMetaRecommender,
+)
+from baybe.searchspace import SearchSpace
+from baybe.targets import NumericalTarget
+from baybe.utils.dataframe import add_fake_results
+
+### Callable function
+
+# This function prints all given parameters and pickles the recommender object to test its serializability.
+
+
+def callable_test(
+    recommender: Optional[SequentialGreedyRecommender] = None,
+    searchspace: Optional[SearchSpace] = None,
+    batch_size: Optional[int] = None,
+    train_x: Optional[DataFrame] = None,
+    train_y: Optional[DataFrame] = None,
+):
+    """Test the functionality of the implemented callable attribute.
+
+    Args:
+        recommender: Recommender to be used.
+        searchspace: The search space from which to recommend the points.
+        batch_size: The number of points to be recommended.
+        train_x: Optional training inputs for training a model.
+        train_y: Optional training labels for training a model.
+    """
+    print("Start callable_test")
+    print(f"""\ntrain x:\n{train_x}\ntrain y:\n{train_y}""")
+    print("End callable_test")
+
+
+### Creating the recommender object
+
+recommender = TwoPhaseMetaRecommender(
+    initial_recommender=RandomRecommender(),
+    recommender=SequentialGreedyRecommender(
+        user_callable=callable_test,
+    ),
+)
+
+### Example Searchspace and objective parameters
+
+# We use the same data used in the [`campaign`](./campaign.md) example.
+
+dict_solvent = {
```

Comment:
as mentioned int he call, a leaner example could be crafted or taken as baseline

Reply:
I removed the unnecessary stuff from this example. Could you please recheck and tell me what do think about it? @AVHopp you too please

Reply:
Looks good to me, will let @Scienfitz resolve.

Reply:
looks good for now